Mention `backend_options` in workflow syntax docs (#5096)
woodpecker-ci/woodpecker: docs/docs/20-usage/20-workflow-syntax.md

Using `directory`, you can set a subdirectory of your repository or an absolute path inside the Docker container in which your commands will run.

### `backend_options`

> Using `backend_options`, you may set options that are specific to the particular backend used to execute the steps. As an example, you may want to set the user and/or group used inside a Docker container; or you may want to specify the service account for Kubernetes.

Reviewer: ```suggestion
With `backend_options` you can define options that are specific to the respective backend that is used to execute the steps. For example, you can specify the user and/or group used in a Docker container or you can specify the service account for Kubernetes.
```

> For more details check the configuration docs of the backend you are using:

Reviewer: ```suggestion
Further details can be found in the documentation of the used backend:
```
